lib/auth/auth_with_roles.go

```diff
@@ -1584,6 +1584,59 @@ func (a *ServerWithRoles) authContextForSearch(ctx context.Context, req *proto.L
 	return extendedContext, nil
 }
 
+// GetSSHTargets gets all servers that would match an equivalent ssh dial request. Note that this method
+// returns all resources directly accessible to the user *and* all resources available via 'SearchAsRoles',
+// which is what we want when handling things like ambiguous host errors and resource-based access requests,
+// but may result in confusing behavior if it is used outside of those contexts.
```

Comment:
Is this `SearchAsRoles` behavior common across all the various things that implement this API? If so, then consider whether we should slap this warning on them as well to avoid misuse down the line.